File: src/main/java/com/hjj/apiserver/repositroy/UserRepository.java
package com.hjj.apiserver.repositroy;

import com.hjj.apiserver.domain.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<UserEntity, Long>, UserRepositoryCustom {
    Optional<UserEntity> findByUserId(String userId);
    Optional<UserEntity> findByProviderAndProviderId(UserEntity.Provider provider, String providerId);
    Boolean existsUserEntityByProviderIdAndProviderAndDeleteYn(String providerId, UserEntity.Provider provider, char deletedYn);
    Boolean existsUserEntityByUserId(String UserId);
    Boolean existsUserEntityByNickNameAndUserNoNot(String nickName, Long userNo);
    Optional<UserEntity> findByRefreshToken(String refreshToken);

}

File: src/main/kotlin/com/hjj/apiserver/repository/user/UserRepository.kt
package com.hjj.apiserver.repository.user

import com.hjj.apiserver.domain.user.Provider
import com.hjj.apiserver.domain.user.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository: JpaRepository<User, Long>, UserRepositoryCustom {
    fun findByUserId(userId: String): User?
    fun findByRefreshToken(refreshToken: String): User?
    fun findByProviderAndProviderId(provider: Provider, providerId: String): User?
    fun findByNickName(nickName: String): User?
    fun existsByProviderIdAndProviderAndDeleteYn(providerId: String, provider: Provider, deletedYn: Char = 'N'): Boolean
}